src/unlessP.js

```diff
@@ -0,0 +1,14 @@
+const always = require('ramda/src/always')
+const curry  = require('ramda/src/curry')
+const ifElse = require('ramda/src/ifElse')
+
+// unlessP :: (a -> Promise Boolean) -> (a -> Promise a) -> a -> Promise a
+const unlessP = (pred, f, x) => {
+  const run = () =>
+    Promise.resolve(x).then(f)
+
+  return Promise.resolve(x).then(pred)
+    .then(ifElse(Boolean, always(x), run))
```

Comment:
I guess you could just do `whenP(complement(pred), f, x)`, if you felt so inclined...

Reply:
I thought about that, but the predicate returns a `Promise`, not a `Boolean`.

Reply:
`complementP`! :troll: